src/main/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStore.kt

```diff
@@ -166,6 +179,205 @@ class ObservationResultsStore(private val dslContext: DSLContext) {
+              }
+            }
+      }
+
+  private val recordedTreesMultiset =
+      with(RECORDED_TREES) {
+        DSL.multiset(
+                DSL.select(
+                        ID,
+                        recordedBranchesMultiset,
+                        DESCRIPTION,
+                        DIAMETER_AT_BREAST_HEIGHT_CM,
+                        HEIGHT_M,
+                        IS_DEAD,
+                        IS_TRUNK,
+                        POINT_OF_MEASUREMENT_M,
+                        SHRUB_DIAMETER_CM,
+                        SPECIES_ID,
+                        SPECIES_NAME,
+                        TREE_GROWTH_FORM_ID,
+                        TREE_NUMBER,
+                    )
+                    .from(this)
+                    .where(OBSERVATION_ID.eq(OBSERVATION_BIOMASS_DETAILS.OBSERVATION_ID))
+                    .and(MONITORING_PLOT_ID.eq(OBSERVATION_BIOMASS_DETAILS.MONITORING_PLOT_ID))
+                    .orderBy(TREE_NUMBER))
+            .convertFrom { result ->
+              result.map {
+                ExistingRecordedTreeModel(
+                    id = it[ID]!!,
+                    branches = it[recordedBranchesMultiset],
+                    description = it[DESCRIPTION],
+                    diameterAtBreastHeightCm = it[DIAMETER_AT_BREAST_HEIGHT_CM],
+                    heightM = it[HEIGHT_M],
+                    isDead = it[IS_DEAD]!!,
+                    isTrunk = it[IS_TRUNK],
+                    pointOfMeasurementM = it[POINT_OF_MEASUREMENT_M],
+                    shrubDiameterCm = it[SHRUB_DIAMETER_CM],
+                    speciesId = it[SPECIES_ID],
+                    speciesName = it[SPECIES_NAME],
+                    treeGrowthForm = it[TREE_GROWTH_FORM_ID]!!,
+                    treeNumber = it[TREE_NUMBER]!!,
+                )
+              }
+            }
+      }
+
+  private val biomassDetailsMultiset =
+      with(OBSERVATION_BIOMASS_DETAILS) {
+        DSL.multiset(
+                DSL.select(
+                        biomassAdditionalSpeciesMultiset,
+                        DESCRIPTION,
+                        FOREST_TYPE_ID,
+                        HERBACEOUS_COVER_PERCENT,
+                        OBSERVATION_ID,
+                        PH,
+                        biomassQuadratDetailsMultiset,
+                        biomassQuadratSpeciesMultiset,
+                        SALINITY_PPT,
+                        SMALL_TREES_COUNT_LOW,
+                        SMALL_TREES_COUNT_HIGH,
+                        SOIL_ASSESSMENT,
+                        MONITORING_PLOT_ID,
+                        TIDE_ID,
+                        TIDE_TIME,
+                        recordedTreesMultiset,
+                        WATER_DEPTH_CM,
+                    )
+                    .from(this)
+                    .where(OBSERVATION_ID.eq(OBSERVATIONS.ID))
+                    .orderBy(MONITORING_PLOT_ID))
+            .convertFrom { result ->
+              result.map {
```

Comment:
Naming the lambda parameter here would make this clearer since there are lambda functions nested inside this one. My general rule of thumb is to only use `it` for the innermost lambda functions in a block, and only in cases where the body of the lambda function is no more than a few lines long.